src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxePsiClass.java

```diff
@@ -450,7 +451,7 @@ else if (this instanceof HaxeEnumDeclaration) { // enum
 
   @Override
   public boolean isPublic() {
-    return (!isPrivate() && super.isPublic()); // do not change the order of- and the- expressions
+    return !isPrivate();
```

Comment:
isPublic always false because of super private - check sources.

Reply:
I don't see that.  This function never calls super.isPrivate() and super.isPublic() never calls isPrivate().

Reply:
HaxeFileModel has method getExposedMembers() as a new feature of this PR.
Inside this method we are getting all class models and checking their "isPublic" value to decide - expose them or not.
In the previous flow: 
```
return !isPrivate() && super.isPublic();
```
Never works right. Just look at conditions in super, regular class like:

```
class ModuleClass {
}
```
would not be public if we'll check it by `isPublic` defined in `AbstractHaxeNamedComponent`, because it's not an extern class, not an interface or anonymous type, and it doesn't have a `public` access property in it's declaration.

Check if it's not a private is sufficient.